gelbooru-overhaul.user.js

```diff
@@ -9,485 +9,1122 @@
-                            });
-                            if (remove) {
-                                elem.remove();
-                            }
-                        } else if (tagFound(blacklistLine)) { // Simple & straightforward blacklisting
-                            elem.remove();
-                        }
-                    });
-                    function tagFound(query) { // Check if a tag is present in the imgElem
-                        var tags = imgElem.alt.split(",");
-                        tags = tags.map(tag => tag.trim())
-                        if (tags.includes(query)) {
-                            return true;
-                        }
-                        return false;
-                    }
-                }
-            });
-            Object.values(document.querySelector(".mainBodyPadding").querySelectorAll("div")).reverse()[1].querySelectorAll("a").forEach((aElem) => {
-                var imgElem = aElem.querySelector("img");
-                if (config.gallery.higherResThumbnailsOnHover) { // Higher-Resolution Preview When Hovering Over Thumbnails
-                    imgElem.addEventListener("mouseenter", function() {
-                        convertThumbnail(imgElem, aElem, false);
-                    }, false);
-                }
-                if (config.gallery.rightClickDownload) { // Download Images in Gallery on Right-Click
-                    imgElem.addEventListener("contextmenu", (event) => {
-                        event.preventDefault();
-                        convertThumbnail(imgElem, aElem, true).then(function() {
-                            downloadImage(imgElem, aElem);
-                        });
-                    })
-                }
-                if (config.gallery.removeTitle) { // Remove Title from Thumbnails
-                    imgElem.title = "";
+
+    /**
+     * @class Class that manages light/dark theme switching
+     */
+    class ThemeManager {
+        isMatchMediaSupported = (window.matchMedia && window.matchMedia('(prefers-color-scheme)').media !== 'not all');
+        constructor() {
+            this.checkForThemeSwitch();
+            this.applyCssThemeVariable();
+
+            if (this.isMatchMediaSupported)
+                window.matchMedia('(prefers-color-scheme: dark)').addEventListener("change", e => this.checkForThemeSwitch());
+
+            this.scheduleCheckForThemeSwitch();
+
+            let darkModeButton = Object.values(document.querySelectorAll("#myTopnav a, #navbar a")).filter(i => i.getAttribute("href").includes("javascript:;"))[0];
+            darkModeButton.onclick = undefined;
+            darkModeButton.setAttribute("title", "Click to force switch dark mode for current session\nRight click to clear force mode");
+            darkModeButton.addEventListener("click", e => this.switchForceSessionMode());
+            darkModeButton.addEventListener("contextmenu", e => this.clearForceSessionMode());
```

Comment:
```suggestion
            let darkModeButton = document.querySelector("#myTopnav a[onclick*='darkModeToggle()'][href*='javascript:;']");
            if (darkModeButton !== null) {
                darkModeButton.onclick = undefined;
                darkModeButton.setAttribute("title", "Click to force switch dark mode for current session\nRight click to clear force mode");
                darkModeButton.addEventListener("click", e => this.switchForceSessionMode());
                darkModeButton.addEventListener("contextmenu", e => this.clearForceSessionMode());
            }
```
Simplify selector for darkModeButton and fix crash on home page and issue tracker that stops page from loading.

Reply:
straightforward fix d953a747f847d9ea5cc6cdf865f326fb9fa5935b

Reply:
7fc9185ef8b53cf9bdae553f0cd872e1e8a9ea4f inspired by the suggested selector